pymeasure/instruments/hp/hp8753e.py

```diff
@@ -0,0 +1,394 @@
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+__version__ = "0.1.1"
+
+import re
+import warnings
+from functools import cached_property
+from io import BytesIO
+from time import sleep
+from time import time as now
+
+import numpy as np
+from pyvisa import VisaIOError
+
+from pymeasure.adapters import PrologixAdapter
+from pymeasure.instruments import Instrument, RangeException, discreteTruncate
+
+
+class HP8753E(Instrument):
+    """Represents the HP8753E Vector Network Analyzer
+    and provides a high-level interface for taking scans of the
+    scattering parameters.
+
+    Template copied from Agilent 8722 class in pymeasure
+    """
+
+    SCAN_POINT_VALUES = [3, 11, 21, 26, 51, 101, 201, 401, 801, 1601]
+    SCATTERING_PARAMETERS = ["S11", "S12", "S21", "S22"]
+    S11, S12, S21, S22 = SCATTERING_PARAMETERS
+
+    start_frequency = Instrument.control(
+        "STAR?",
+        "STAR %e Hz",
+        """ A floating point property that represents the start frequency
+        in Hz. This property can be set.
+        """,
+    )
+    stop_frequency = Instrument.control(
+        "STOP?",
+        "STOP %e Hz",
+        """ A floating point property that represents the stop frequency
+        in Hz. This property can be set.
+        """,
+    )
+    sweep_time = Instrument.control(
+        "SWET?",
+        "SWET%.2e",
+        """ A floating point property that represents the sweep time
+        in seconds. This property can be set.
+        """,
+    )
+    averages = Instrument.control(
+        "AVERFACT?",
+        "AVERFACT%d",
+        """ An integer representing the number of averages to take. Note that
+        averaging must be enabled for this to take effect. This property can be set.
+        """,
+        cast=lambda x: int(float(x)),  # need float() to convert scientific notation in strings
+    )
+    averaging_enabled = Instrument.control(
+        "AVERO?",
+        "AVERO%d",
+        """ A bool that indicates whether or not averaging is enabled. This property
+        can be set.""",
+        cast=bool,
+    )
+
+    correction_enabled = Instrument.control(
+        "CORR?",
+        "CORR%d",
+        """ A bool that indicates whether or not correction is enabled. This property
+        can be set.""",
+        cast=bool,
+    )
+
+    power = Instrument.control(
+        "POWE?",
+        "POWE%.2e",
+        """ A float that can be set from -70 to +10dBm to control output power from the VNA ports. 
+        This property can be set.""",
+        cast=lambda x: float(x),
```

Comment:
`lambda x: float(x)` is exactly the same as `float`, which is the default value for the `cast` parameter, such that you do not have to use it at all.
```suggestion
```